rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -445,7 +441,7 @@ else if (isCached.get(idx).getLocation() == Location.CACHED) {
                 affixBuffer = new byte[blockSize];
               }
               if (remoteReadRequestChain == null) {
-                remoteReadRequestChain = new RemoteReadRequestChain(getParentDataInputStream(), localPath, directWriteBuffer, affixBuffer);
+                remoteReadRequestChain = ReadRequestChainFactory.createReadRequestChain(RemoteReadRequestChain.class, getParentDataInputStream(), localPath, directWriteBuffer, affixBuffer, null);
```

Comment:
Same as above

Reply:
Do this for all places. Keep the logic of passing optional arguments be in Factory for all Chains

Reply:
Done